### DISCUSSION TITLE
Possible to get client type with ReactPHP sockets?

### DISCUSSION THREAD

**Discussion by briantknies** [reactphp/reactphp#499] [Q&A] [answered]
Hi. I was just wondering if it's possible to get the type of client connected to a socket server using ReactPHP sockets. For example, if the user connected with MSDOS, is it possible to retrieve that the user connected through MSDOS programmatically? I tried dumping all the class methods of the ConnectionInterface object to look for hints but didn't find anything. I'm not sure this is possible with PHP but figured I'd ask. Thanks.

**Comment by WyriHaximus** [comment, marked as answer]
The only thing we get is the remote IP and port a connection comes from. Unless there is something optionally available in TCP IP that would provide this information, but we aren't aware of that. The simplest way would be for the client to send you that data.

**Reply by SimonFrings** [reply]
I agree with @WyriHaximus, the client needs to send this data in order for your server to get this information, I would also add that getting this information by default could violate data protection. If you're using HTTP you can access the sent `User-Agent` HTTP header in order to get more information about the client/user, quoting from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent:

> The User-Agent [request header](https://developer.mozilla.org/en-US/docs/Glossary/Request_header) is a characteristic string that lets servers and network peers identify the application, operating system, vendor, and/or version of the requesting [user agent](https://developer.mozilla.org/en-US/docs/Glossary/User_agent).

Hope this helps.

**Reply by briantknies** [reply]
That makes sense. I didn't know if that information was being passed somehow. I'll look into seeing if the client can pass that info. Thank you both for the quick response.
